lib/hedwig/robot.ex

```diff
@@ -69,51 +69,57 @@ defmodule Hedwig.Robot do
 
       def log(msg) do
         Logger.unquote(@log_level)(fn ->
-          "#{inspect msg}"
+          msg
         end, [])
       end
 
       def __adapter__, do: @adapter
 
       def init({robot, opts}) do
-        opts = Keyword.merge(robot.config, opts)
-        {:ok, adapter} = @adapter.start_link(robot, opts)
-
-        {aka, opts}   = Keyword.pop(opts, :aka)
-        {name, opts}  = Keyword.pop(opts, :name)
-        responders    = Keyword.get(opts, :responders, [])
+        opts = robot.config(opts)
+        {aka, opts} = Keyword.pop(opts, :aka)
+        {name, opts} = Keyword.pop(opts, :name)
+        {responders, opts} = Keyword.pop(opts, :responders, [])
 
         unless responders == [] do
           GenServer.cast(self, :install_responders)
         end
 
+        {:ok, adapter} = @adapter.start_link(robot, opts)
+        {:ok, responder_sup} = Hedwig.Responder.Supervisor.start_link()
+
         {:ok, %Hedwig.Robot{
           adapter: adapter,
           aka: aka,
           name: name,
           opts: opts,
-          pid: self()}}
+          responder_sup: responder_sup,
+          responders: responders}}
       end
 
       def handle_connect(state) do
-        Logger.warn """
-        #{inspect __MODULE__}.handle_connect/1 default handler invoked.
-        """
         {:ok, state}
       end
 
       def handle_disconnect(_reason, state) do
-        Logger.warn """
-        #{inspect __MODULE__}.handle_disconnect/2 default handler invoked.
-        """
         {:reconnect, state}
       end
 
       def handle_in(msg, state) do
-        Logger.warn """
-        #{inspect __MODULE__}.handle_in/2 default handler invoked.
-        """
-        {:ok, state}
+        case msg do
+          %Hedwig.Message{} ->
+            {:dispatch, msg, state}
+          _ ->
+            {:noreply, state}
+        end
+      end
```

Comment:
Yeah, I do like this a bit better. Also...could add these to the Robot generator.